test/integration/getserversideprops/test/index.test.js

```diff
@@ -260,6 +260,46 @@ const runTests = (dev = false) => {
     expect(data.pageProps.params).toEqual({ path: ['first'] })
   })
 
+  it('should have original req.url for /_next/data request dynamic page', async () => {
+    const curUrl = `/_next/data/${buildId}/blog/post-1.json`
+    const data = await renderViaHTTP(appPort, curUrl)
+    const { appProps } = JSON.parse(data)
+
+    expect(appProps).toEqual({
+      url: curUrl,
+      query: { post: 'post-1' },
+      asPath: curUrl,
+      pathname: '/blog/[post]',
+    })
+  })
+
+  it('should have original req.url for /_next/data request', async () => {
```

Comment:
This wouldn't work in serverless I believe 🤔

Reply:
Serverless mode is being tested here, which part doesn't seem like it would work?

Reply:
I looked at it the wrong way 👍